docs: Environment Automation Doc (#3854)
kubeshop/tracetest: docs/docs/concepts/environment-automation.md

# Environment Automation

## Resource Definition

> The Tracetest UX provides several options for maintaining Resource Definitions for Tracetest constructs.

Reviewer: ```suggestion
The Tracetest Web UI provides several options for maintaining Resource Definitions for Tracetest constructs.
```

On the right are the automate commands for applying this resource in the CLI.

### Trace Polling

Similarly, on the **Trace Polling** tab, you will find the **Resource Definition** link for each polling profile in the trace:

![Trace Polling Resource Definition](../img/trace-polling-resource-definition.png)

Selecting the **Definition** link will open the **Resource Definition** modal:

![Trace Polling Resource Definition Modal](../img/trace-polling-resource-definition-modal.png)

### Demo

On the **Demo** tab, you will find link to work with resource definitions, as well:

![Demo Resource Definition](../img/demo-resource-definition.png)

### Analyzer

On the **Analyzer** tab, click on **Resource Definition**:

![Analyzer Resource Definition](../img/analyzer-resource-definition.png)

The analyzer modal shows one single action.

![Analyzer Resource Definition Modal](../img/analyzer-resource-definition-modal.png)

### Test Runner

On the **Test Runner** tab, click on **Resource Definition**:

![Test Runner Resource Definition](../img/test-runner-resource-definition.png)

![Test Runner Resource Definition Modal](../img/test-runner-resource-definition-modal.png)

### Variable Sets

> In the navigation panel on the left of the screen, click on **Variable Sets**:

Reviewer: ```suggestion
In the navigation panel on the left of the screen, click on **Variable Sets**.
```

> ![Variable Set Resource Definition](../img/test-runner-resource-definition.png)

Reviewer: ```suggestion

```

Reviewer: Removing this image since it's not super necessary.
